packages/headless/src/features/standalone-search-box-set/standalone-search-box-set-actions.ts

```diff
@@ -129,5 +130,8 @@ export const buildPlanRequest = (
     ...(state.context && {context: state.context.contextValues}),
     ...(state.pipeline && {pipeline: state.pipeline}),
     ...(state.searchHub && {searchHub: state.searchHub}),
+    ...(state.configuration.analytics.enabled && {
+      visitorId: await getVisitorID(),
```

Comment:
I'm not a fan of how much this change is leaking. I think there might be an approach that better fits with the project,

Could we dispatch an async thunk on engine init that does what `getVisitorID` is doing? The action would be handled by a reducer that writes to a property in the headless store, allowing us to operate synchronously.

The dispatch might happen somewhere [around here](https://github.com/coveo/ui-kit/blob/master/packages/headless/src/app/engine.ts#L151) ...

Reply:
This would not correctly handle if the visitor ID changes. 

If there's any (internal or external) code that modifies the value in cookie/async storage for any reason that we might not have foreseen today, then there would be a desync with Headless.

Reply:
That's true ... alright, maybe this is the best we can do